contracts/validators/CredentialAtomicQueryValidator.sol

```diff
@@ -6,79 +6,100 @@ import {GenesisUtils} from "../lib/GenesisUtils.sol";
 import {ICircuitValidator} from "../interfaces/ICircuitValidator.sol";
 import {IVerifier} from "../interfaces/IVerifier.sol";
 import {IState} from "../interfaces/IState.sol";
+import {PoseidonFacade} from "../lib/Poseidon.sol";
 
 abstract contract CredentialAtomicQueryValidator is OwnableUpgradeable, ICircuitValidator {
-    IVerifier public verifier;
+    struct CredentialAtomicQuery {
+        uint256 schema;
+        uint256 claimPathKey;
+        uint256 operator;
+        uint256 slotIndex;
+        uint256[] value;
+        uint256 queryHash;
+        uint256[] allowedIssuers;
+        string[] circuitIds;
+        bool skipClaimRevocationCheck;
+    }
+
+    // This empty reserved space is put in place to allow future versions
+    // of the CredentialAtomicQuerySigValidator contract to inherit from other contracts without a risk of
+    // breaking the storage layout. This is necessary because the parent contracts in the
+    // future may introduce some storage variables, which are placed before the CredentialAtomicQuerySigValidator
+    // contract's storage variables.
+    // (see https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#storage-gaps)
+    // slither-disable-next-line shadowing-state
+    // slither-disable-next-line unused-state
+    uint256[500] private __gap_before;
```

Comment:
Is it the best approach to input 500 slots gap if the class is guaranteed not to be the last in the inheritance chain? The same question for the ZKPVerifier.

Reply:
what is your suggestion

Reply:
CredentialAtomicQuerySigValidator, CredentialAtomicQueryMTPValidator and ZKPVerifier are last in the inheritance chain.

Reply:
As we discussed we adding this gap_before to parent contract CredentialAtomicQueryValidator if some changes in OwnableUpgradeable and not adding to child MTP/Sig